code/drasil-lang/Language/Drasil/Chunk/Constrained.hs

```diff
@@ -50,6 +50,33 @@ instance Eq            ConstrainedChunk where c1 == c2 = (c1 ^. qd . uid) == (c2
 instance MayHaveUnit   ConstrainedChunk where getUnit = getUnit . view qd
 -- ^ Finds units contained in the 'QuantityDict' used to make the 'ConstrainedChunk'.
 
+data ConstrainedQDef = ConstrainedQDef { _qd' :: QuantityDict
+                                       , _constr' :: [Constraint] -- should really be NEList
+                                       }
+
+makeLenses ''ConstrainedQDef                          
+
+instance HasUID        ConstrainedQDef   where uid = qd' . uid
+-- ^ Finds 'UID' of the 'QuantityDict' used to make the 'ConstrainedQDef  '.
```

Comment:
Note that a recent PR from Anthony has moved all these comments from below to above the `instance`, as I think Haddock (the documentation generator) prefers that. Please coordinate.

Reply:
Okay